esp-hal-common/src/soc/esp32c3/efuse.rs

```diff
@@ -66,23 +61,28 @@ impl Efuse {
     }
 
     /// Get version of RTC calibration block
-    pub fn get_rtc_calib_version() -> u32 {
+    ///
+    /// see https://github.com/espressif/esp-idf/blob/7052a14d6/components/efuse/esp32c3/esp_efuse_rtc_calib.c#L12
+    pub fn get_rtc_calib_version() -> u8 {
         let efuse = unsafe { &*EFUSE::ptr() };
 
-        efuse.rd_sys_part1_data4.read().sys_data_part1_4().bits() & 0x7
+        // see https://github.com/espressif/esp-idf/blob/7052a14d6/components/efuse/esp32c3/esp_efuse_table.csv#L143
+        (efuse.rd_sys_part1_data4.read().bits() & 0x3) as _
```

Comment:
I personally don't like reading Efuses this low level. I think the abstraction [here](https://github.com/dimpolo/esp-hal/commit/f79432469f60df636e00ad80077e27f7b21f9c14#diff-c262537b65a1612b4c12352b3dcfdf9dadbbe6552dedf6250480aea2082071edR213) is easier to read, write, and compare to esp-idf data.

I'm not sure what you think about it, and why you chose reading the registers directly. They are probably faster and they may generate less code, but I don't think this is very critical.

I can refactor the efuses later if you don't have strong feelings against it.

Reply:
Currently we lacks unified way of getting fields from efuse.
The existing methods such as `Efuse::get_mac_address` `Efuse::get_flash_encryption` also works via PAC.